sql/analyzer/fix_field_indexes.go

```diff
@@ -0,0 +1,110 @@
+package analyzer
+
+import (
+	"github.com/src-d/go-mysql-server/sql"
+	"github.com/src-d/go-mysql-server/sql/expression"
+	"github.com/src-d/go-mysql-server/sql/plan"
+)
+
+// fixFieldIndexesOnExpressions executes fixFieldIndexes on a list of exprs.
+func fixFieldIndexesOnExpressions(schema sql.Schema, expressions ...sql.Expression) ([]sql.Expression, error) {
+	var result = make([]sql.Expression, len(expressions))
+	for i, e := range expressions {
+		var err error
+		result[i], err = fixFieldIndexes(schema, e)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return result, nil
+}
+
+// fixFieldIndexes transforms the given expression setting correct indexes
```

Comment:
"the given expression setting correct indexes for GetField"

 wat?

Reply:
Done